src/ConfigurationClientWrapper.ts

```diff
@@ -21,11 +21,17 @@ export class ConfigurationClientWrapper {
 
     updateBackoffStatus(successfull: boolean) {
         if (successfull) {
-            this.#failedAttempts = 0;
+            if (this.failedAttempts > 0) {
+                this.failedAttempts = 0;
+            }
+            this.failedAttempts -= 1;
```

Comment:
failedAttempts can be negative. This doesn't make sense. If this behavior is only for testing, don't do that.